### DISCUSSION TITLE
eoapi/raster service constantly crashes and data ingestion not straight forward during deployment.

### DISCUSSION THREAD

**Discussion by mdominguezd** [developmentseed/k8s-eoapi#82] [Q&A]
Hey there,

I am quite new using Docker and Kubernetes. I tried deploying the eoApi in a Kubernetes cluster with the helm chart provided, but already had some issues. 

1. Firstly, the raster service deployed constantly shows error 503: service unavailable. And the pod shows a CrashLoopBackOff error that I don't know how to solve.
2. Moreover, for the data ingestion the pypgstac version (0.7.0) installed in the stac-eoapi pod was not compatible with the version of the database (0.8.2) . To solve this I installed a newer version of pypgstac which threw me some pip dependencies errors: 
``` 
stac-fastapi-pgstac 2.4.9 requires pypgstac==0.7.*, but you have pypgstac 0.8.0 which is incompatible. 
fastapi 0.97.0 requires pydantic!=1.8,!=1.8.1,<2.0.0,>=1.7.4, but you have pydantic 2.7.3 which is incompatible.
```
Regardless of those errors I was then able to ingest my collections and items into the stac API.

Any help I could get with both topics, I would highly appreciate it.

Cheers

**Comment by ranchodeluxe** [comment]
Thanks for writing us @mdominguezd and sorry for the late reply ✨ 

>     1. Firstly, the raster service deployed constantly shows error 503: service unavailable. And the pod shows a CrashLoopBackOff error that I don't know how to solve.

The `/raster` service is probably running out of memory and killing the pod. We have integration tests that run on GCP and they use these resource limits for a very small node: https://github.com/developmentseed/eoapi-k8s/blob/main/helm-chart/eoapi/test-gcp-unittest-values.yaml#L32-L38

But then again, if your node is too small for the required resources then the pod will not be able to be scheduled but that should be apparent in the event logs if you `kubectl describe pod/<raster-pod>`

>     2. Moreover, for the data ingestion the pypgstac version (0.7.0) installed in the stac-eoapi pod was not compatible with the version of the database (0.8.2) . To solve this I installed a newer version of pypgstac which threw me some pip dependencies errors:
> 

Thanks for catching this,t hat was fixed a bit ago and should be fine in `main` now
